src/atoms/referencesState.ts

```diff
@@ -36,6 +36,50 @@ const removeReferenceAtom = atom(null, (get, set, id: string) => {
   set(setReferencesAtom, newReferences);
 });
 
+const UPDATABLE_FIELDS: (keyof ReferenceItem)[] = ['citationKey', 'title', 'publishedDate', 'authors'];
+export const updateReferenceAtom = atom(null, async (get, set, id: string, updatedReference: ReferenceItem) => {
+  const reference = get(getDerivedReferenceAtom(id));
+  if (!reference) {
+    console.warn('Cannot find reference with ID', id);
+    return;
+  }
+
+  const patch: Partial<ReferenceItem> = UPDATABLE_FIELDS.reduce(
+    (acc, field) => ({
+      ...acc,
+      ...patchEntry(field, reference, updatedReference),
+    }),
+    {},
+  );
+
+  if (Object.keys(patch).length === 0) {
+    console.log('No change detected.');
+    return;
+  }
+
+  // Call backend (patch of updatable fields)
+  await updateReference(reference.filename, patch);
+
+  // Update local atoms
+  const references = get(getReferencesAtom);
+  const updatedReferences = {} as ReferencesState;
```

Comment:
Could you write `const updatedReferences: ReferencesState = {}` instead? I think it's better to avoid `as` when possible